src/file_upload.rs

```diff
@@ -0,0 +1,106 @@
+use actix_web::{
+    dev, error,
+    http::header::{ContentDisposition, LOCATION},
+    multipart, Error, FutureResponse, HttpMessage, HttpRequest, HttpResponse,
+};
+use futures::{future, Future, Stream};
+use std::{
+    io::Write,
+    path::{Component, PathBuf},
+};
+
+pub fn save_file(
+    field: multipart::Field<dev::Payload>,
+    file_path: PathBuf,
+) -> Box<Future<Item = i64, Error = Error>> {
+    let mut file = match std::fs::File::create(file_path) {
```

Comment:
Can't you use `?` here?

Reply:
As far as I know using `?` in function returning future is not possible.

Reply:
True. I just checked and sadly it's not possible.